API_README.md

```diff
@@ -4,6 +4,36 @@
 
 Easy-to-use JSON-based REST API for OpenEMR. All code is done in classes and separate from the view to help with codebase modernization efforts. FHIR is also supported, see FHIR API documentation [here](FHIR_README.md)
 
+## Implementation
+
+REST API endpoints are defined in the [primary routes file](_rest_routes.inc.php). The routes file maps an external, addressable
+endpoint to the OpenEMR controller which handles the request, and also handles the JSON data conversions.
+
+```php
+"POST /api/patient" => function () {
+    RestConfig::authorization_check("patients", "demo");
+    $data = (array)(json_decode(file_get_contents("php://input")));
+    return (new PatientRestController())->post($data);
+}
+```
+
+At a high level, the request processing flow consists of the following steps:
+```bash
+JSON Request -> Controller Component -> Validation -> Service Component -> Database
+```
+
+The logical response flow begins with the database result:
+```bash
+Database Result -> Service Component -> Controller Component -> RequestControllerHelper -> JSON Response
+```
```

Comment:
Just a little opinion, Maybe change the `bash` to `sh` will be more consistent with the current document :)

Reply:
@stu01509 I've updated this in #3453. Thanks!

Reply:
I Got it 😀